### ISSUE TITLE
Please help!!!

### ISSUE THREAD

**Issue by kyliejane1** [vmkernel/expense-iq#4]
Hi there, you will be an absolute lifesaver if I can get this solution of yours to work - this happened to me about a week ago, and I've come across this in trying to find some way to fix it. I have 5 years of past data and a load of future data that I'm desperate to restore, and have heard zero, nothing at all, from my 6 or 7 emails to handyappsforlife. Absolutely disgraceful, I'm so angry about this!!!! If they're not going to fix the bug, there should be a very very clear warning to not use apostrophes in the database, there should be a warning about this!!!

Anyway, I have very little experience in anything code related but am prepared to try anything to get this data restored - your instructions to beowulf83 were extremely helpful, that's the level of instruction that I need! I have however, run into this issue with loading the script:

![Screenshot 2023-09-22 065457](https://github.com/vmkernel/expense-iq/assets/145719106/2a39296c-83b0-4908-8b3b-a072f2dd9ac2)

Any assistance would be greatly appreciated, I feel very close to finally being able to get my data back, so am more hopeful than I've been for the last week anyway! The error in mine will not be just one line, there is a payee that I regularly use, that has an apostrophe in the name, so I assume that it's going to be a long and tedious search and delete mission if I can get the file decoded, but which I'm happy to spend the time on, just to have it restored!!!!

**Comment by vmkernel** [comment]
Hi!
That's a very strange behavior from PowerShell...

It might be caused by some kind of "untrusted source" flag in the properties of the downloaded script. Please check the properties of the script in Windows Explorer, and if you'll find that kind of flag on the General tab, please uncheck it. You can also post the screenshot here.

If it doesn't work, try to set execution policy to "Unrestricted" (instead of remote signed) and restart PowerShell console.

**Comment by kyliejane1** [comment]
Hey that worked perfectly - it appears it was blocked by default and I checked the box to unblock it :) My problem now is that I have decoded and recoded and my data still won't reload into the app!! I have a feeling that it's because I have used so many 'special characters' throughout, I delete a batch of what I think are special characters [Sublime Text is brilliant!] recode it and try again. No luck and repeat with deleting others. No luck and repeat. Etc etc. Logcat has not really been of much use, it found one -probably the first one it tried to load, based on where it was placed in the decoded file - in the format that I've seen in your example and beowulf83's example of logcat files of this error, and then my device appears to have just given up logging them! I'm not really sure what are considered 'special characters' - for example are currency symbols that I enter in descriptions special characters? Are accented letters? Would you know where I'm able to get a list of what this app's code considers special characters? So sorry to be a pain, but you're my only hope right now at restoring my data, and I haven't given up just yet!! Thank you so much again for any help :)

**Comment by vmkernel** [comment]
Hi! And don't give up! ))

Logcat can show only the first failed line in a file (at least in case of Expense IQ) as it's actually an error called Unhanded Exception, that means the app (or one of its parts) crashes and unable to log anything further.

Unfortunately, I can't tell what is special symbols from the Expense IQ point of view. So the only way is to fix errors one by one.

Could you show the current logcat output?

Please, note, that in case of apostrophe symbol (') it's required to remove the only excess symbol(s) and keep the symbols in the start and the end of the string line.

**Comment by kyliejane1** [comment]
I'm thrilled to report that I got it all back 👍😊😊 I started again with the original decoded file and just took the ' and " out, happened to come across [in the 16 846 lines !!] an instance where I had put quotation marks around a word, and that hadn't come up in any of the previous searches that I had done looking for all of the 's instances - got rid of that, crossed my fingers that I had done that only once, as it wasn't easily searchable given the volume of apostrophes that are in the strings themselves! and reloaded it - SUCCESS at last!! SUCH a relief!!!!!! So for me it reloaded with accented letters, & ? / , currency symbols, everything. It really does appear to only be 'and ". I'm still so angry at these people for failing to correct this and then completely ignoring me when I reached out numerous times for assistance with an app that I PAID them for, just a disgraceful amateur company clearly. But thank YOU so much for all of your help, and of course for your wonderful code that got my data back 😊😊
